components.go

```diff
@@ -133,7 +133,7 @@ func (event *VEvent) SetLocation(s string, props ...PropertyParameter) {
 }
 
 func (event *VEvent) SetGeo(lat interface{}, lng interface{}, props ...PropertyParameter) {
-	event.SetProperty(ComponentPropertyGeo, ToText(fmt.Sprintf("%v;%v", lat, lng)), props...)
+	event.SetProperty(ComponentPropertyGeo, fmt.Sprintf("%v;%v", lat, lng), props...)
```

Comment:
Is it safe? The lat parameter is interface{}, but shouldn't it be float? I think, you can inject another event(s) to the serialized calendar with a malicious string lat or lng parameter here.

Reply:
1. It is common for lat and lng to be a string or float64. It should accept both.
2. who's going to inject malicious strings? You can already inject anything you want using `SetProperty`.

Reply:
Where does the parameters come from? From a database? From a user interface? From an API? You don't know.
I think this is a library for creating correct ical files, so it should validate it's input, report if one of your parameter is broken, and it should provide an interface what is easy to use without bugs or security issues. Otherwise we want to use just an ad-hoc stringbuilder for creating ical files.
In my opinion it should accept only floats, and let the lib's user do the string->float conversion and error handling. (Or the function can have a string version, but it should return with the conversions error)

Reply:
Yeah. Sorry been very busy.. I am a fan of strongly typed stuff.  It seems I rushed my review of #17 etc.. I don't believe the solution here is to use the ToText function as that seems to escape the ';' as said above, which according to the very front page of https://icalendar.org/ which is the cited source for this chain of change requests implies it's unescaped. 

I am pretty unhappy about the use of interface{} to be honest. I would like to see a PR address that. I would probably either type switch or create multiple functions if I were doing the change. (Technically fmt.Sprintf's fmt.Formatter is doing that. but it's slow less layers of code is better.)

Okay a quick look at the RFC specifies:
```
3.8.1.6.  Geographic Position

   Property Name:  GEO

   Purpose:  This property specifies information related to the global
      position for the activity specified by a calendar component.

   Value Type:  FLOAT.  The value MUST be two SEMICOLON-separated FLOAT
      values.

   Property Parameters:  IANA and non-standard property parameters can
      be specified on this property.

   Conformance:  This property can be specified in "VEVENT" or "VTODO"
      calendar components.

   Description:  This property value specifies latitude and longitude,
      in that order (i.e., "LAT LON" ordering).  The longitude
      represents the location east or west of the prime meridian as a
      positive or negative real number, respectively.  The longitude and
      latitude values MAY be specified up to six decimal places, which
      will allow for accuracy to within one meter of geographical
      position.  Receiving applications MUST accept values of this
      precision and MAY truncate values of greater precision.

      Values for latitude and longitude shall be expressed as decimal
      fractions of degrees.  Whole degrees of latitude shall be
      represented by a two-digit decimal number ranging from 0 through
      90.  Whole degrees of longitude shall be represented by a decimal
      number ranging from 0 through 180.  When a decimal fraction of a
      degree is specified, it shall be separated from the whole number
      of degrees by a decimal point.

      Latitudes north of the equator shall be specified by a plus sign
      (+), or by the absence of a minus sign (-), preceding the digits
      designating degrees.  Latitudes south of the Equator shall be
      designated by a minus sign (-) preceding the digits designating
      degrees.  A point on the Equator shall be assigned to the Northern
      Hemisphere.

      Longitudes east of the prime meridian shall be specified by a plus
      sign (+), or by the absence of a minus sign (-), preceding the
      digits designating degrees.  Longitudes west of the meridian shall
      be designated by minus sign (-) preceding the digits designating
      degrees.  A point on the prime meridian shall be assigned to the
      Eastern Hemisphere.  A point on the 180th meridian shall be
      assigned to the Western Hemisphere.  One exception to this last
      convention is permitted.  For the special condition of describing
      a band of latitude around the earth, the East Bounding Coordinate
      data element shall be assigned the value +180 (180) degrees.

      Any spatial address with a latitude of +90 (90) or -90 degrees
      will specify the position at the North or South Pole,
      respectively.  The component for longitude may have any legal
      value.
      With the exception of the special condition described above, this
      form is specified in [ANSI INCITS 61-1986].

      The simple formula for converting degrees-minutes-seconds into
      decimal degrees is:

      decimal = degrees + minutes/60 + seconds/3600.

   Format Definition:  This property is defined by the following
      notation:

       geo        = "GEO" geoparam ":" geovalue CRLF

       geoparam   = *(";" other-param)

       geovalue   = float ";" float
       ;Latitude and Longitude components

   Example:  The following is an example of this property:

       GEO:37.386013;-122.082932

```
https://tools.ietf.org/html/rfc5545#section-3.8.1.6

Which sort of settles the matter, it also means the parser side of this will need to be made aware that + might be used.

So do I get support for converting the interface{} to float64 and declining this pr?

Reply:
Seems I misread the diff. I'm going to accept this as it does the right thing. Sorry!